viz/LaserLine.cpp

```diff
@@ -2,44 +2,150 @@
 #include "LaserLine.hpp"
 
 using namespace vizkit3d;
+using namespace vizkit3d_laser_line;
 
-struct LaserLine::Data {
-    // Copy of the value given to updateDataIntern.
-    //
-    // Making a copy is required because of how OSG works
-    base::samples::LaserScan data;
-};
 
+static inline double milimeters_to_meters(x)  { return x / 1000.0; }
 
 LaserLine::LaserLine()
-    : p(new Data)
+    : color(osg::Vec4(0.0, 1.0, 0.0, 0.0))
+    , vertices(new osg::Vec3Array())
+    , lineHolder(new LineHolder())
```

Comment:
You allocate LineHolder but do not delete it in the destructor. A good practice is to look at all "naked" (i.e. non-smart) pointers declared as attributes of your class and making sure they are (1) initialized to something or NULL in the constructor and (2) deleted in the destructor.

Another question you must ask yourself is: do I need to allocate on the heap ? In this case, I am not sure about it. Allocating "by value" instead of "on the heap" is always safer, do it whenever you can.